src/Aspire.Dashboard/Components/Controls/ApplicationName.razor

```diff
@@ -0,0 +1,54 @@
+﻿@namespace Aspire.Dashboard.Components
+@using Aspire.Dashboard.Model
+@using System.Globalization
+@inject IDashboardClient DashboardClient
+
+@code {
+    [Parameter]
+    public required string ResourceName { get; init; }
+
+    [Parameter]
+    public required IStringLocalizer Loc { get; init; }
```

Comment:
Why isn't this injected?

Reply:
Because each caller wants to log this under their own logger, which has a specialized type (e.g. `StringLocalizer<MetricsResources>`). A different instance is injected into each caller.

Reply:
Got it.

Reply:
An alternative would be to replace the resource name and localizer with `Func<string, string>` as a parameter. The formatting happens in the func.

But being explicit is probably better.

Reply:
Wouldn't that be more verbose for the callers?

Reply:
It's probably about the same once the requirement to pass in a LOC is removed.

A func would give more flexibility, but we don't need flexibility, which is why what is happening now is fine.